modules/auxiliary/dos/dns/bind_tsig_badtime.rb

```diff
@@ -64,7 +64,7 @@ def payload
     query << "\x00\x01"  # Class: IN (0x0001)
 
     # Additional records. Name
-    query << "\x0a" + 'local-ddns'
+    query << "\x0alocal-ddns"
```

Comment:
this is definitely uglier...totally understand why it was done like this in the first place, I still think what rubocop has done is fine though

Reply:
Yeah, for the most part this rule is great - it helps fix the Java-style concatenation. Sometimes it's not great for our use cases 🤔 

If it starts becoming a pain, I'm happy to revert this - or make changes so if there's bytes in the strings, it'll skip the concatenation.

For full context, this was originally this was actually rewritten to:

```diff
--- a/modules/auxiliary/dos/dns/bind_tsig_badtime.rb
+++ b/modules/auxiliary/dos/dns/bind_tsig_badtime.rb
@@ -64,7 +64,7 @@ class MetasploitModule < Msf::Auxiliary
     query << "\x00\x01"  # Class: IN (0x0001)
 
     # Additional records. Name
-    query << "\x0a" + 'local-ddns'
+    query << "\nlocal-ddns"
     query << "\x00"
 
     query << "\x00\xfa" # Type: TSIG (Transaction Signature) (250)
@@ -73,7 +73,7 @@ class MetasploitModule < Msf::Auxiliary
     query << "\x00\x1d" # Data length: 29
 
     # Algorithm Name
-    query << "\x0b" + 'hmac-sha256' # The algorithm for local-ddns is hmac-sha256
+    query << "\vhmac-sha256" # The algorithm for local-ddns is hmac-sha256
     query << "\x00"
 
     # Rest of TSIG
@@ -86,8 +86,8 @@ class MetasploitModule < Msf::Auxiliary
   end
 
   def get_domain
-    domain = "\x06" + Rex::Text.rand_text_alphanumeric(6)
-    org = "\x03" + Rex::Text.rand_text_alphanumeric(3)
+    domain = "\u0006#{Rex::Text.rand_text_alphanumeric(6)}"
+    org = "\u0003#{Rex::Text.rand_text_alphanumeric(3)}"
     domain + org
   end
```

But I tweaked it back to what it looked like previously. I imagine this will come up again though. Hm.

Reply:
PR to drop this rule
https://github.com/rapid7/metasploit-framework/pull/14814